includes/Modules/Analytics.php

```diff
@@ -1183,57 +1170,32 @@ protected function setup_services( Google_Site_Kit_Client $client ) {
 	 * Verifies that user has access to the property found in the existing tag.
 	 *
 	 * @since 1.0.0
+	 * @since n.e.x.t Simplified to return a boolean and require account ID.
 	 *
 	 * @param string $property_id Property found in the existing tag.
-	 * @param string $account_id  Optional. Account ID the property belongs to, if known. Default empty.
-	 * @return mixed False if user has no access to the existing property or array with account id and property found.
+	 * @param string $account_id  Account ID the property belongs to.
+	 * @return bool True if the user has access, false otherwise.
 	 */
-	protected function has_access_to_property( $property_id, $account_id = '' ) {
-		if ( empty( $property_id ) ) {
+	protected function has_access_to_property( $property_id, $account_id ) {
+		if ( empty( $property_id ) || empty( $account_id ) ) {
 			return false;
 		}
 
-		if ( ! empty( $account_id ) ) {
-			$account_ids = array( $account_id );
-		} else {
-			$accounts = $this->get_data( 'accounts-properties-profiles' );
-			if ( is_wp_error( $accounts ) || empty( $accounts ) ) {
-				return false;
-			}
-
-			$account_ids = array_map(
-				function( $account ) {
-					return $account->getId();
-				},
-				$accounts['accounts']
-			);
+		// Try to get properties for that account.
+		$properties = $this->get_data( 'properties-profiles', array( 'accountID' => $account_id ) );
+		if ( is_wp_error( $properties ) ) {
+			// No access to the account.
+			return false;
 		}
 
-		$response = false;
-
-		foreach ( $account_ids as $account_id ) {
-			$properties = $this->get_data( 'properties-profiles', array( 'accountID' => $account_id ) );
-			if ( is_wp_error( $properties ) ) {
-				continue;
+		// Ensure there is access to the property.
+		$property_match = array_filter(
+			$properties['properties'],
+			function( $property ) use ( $property_id ) {
+				return $property->getId() === $property_id;
 			}
-
-			$existing_property_match = array_filter(
-				$properties['properties'],
-				function( $property ) use ( $property_id ) {
-					return $property->getId() === $property_id;
-				}
-			);
-
-			if ( ! empty( $existing_property_match ) ) {
-				$response = array(
-					'accountID'  => $account_id,
-					'propertyID' => $property_id,
-				);
-				break;
-			}
-		}
-
-		return $response;
+		);
+		return ! empty( $property_match );
```

Comment:
Since we're only looking for a single match (there should only be one anyways) there's no need or reason to continue iterating over others. I realize this was how it was before, but could be simplified with a `foreach` and `return true` on match, with a `return false` at the very end if there's no match. This is somewhat of a micro-optimization, that may be or may not be more clear, so I'm also fine with keeping it the way it is if you'd like.